memory-bank/research/2026-01-06_plan-mode-implementation.md

```
@@ -0,0 +1,338 @@
+- **File:** `src/tunacode/constants.py:70-80`
+```python
+READ_ONLY_TOOLS = [read_file, grep, list_dir, glob, react, research_codebase, web_fetch]
+WRITE_TOOLS = [write_file, update_file]
+EXECUTE_TOOLS = [bash]
+```
+
+### 3. Authorization Rule Chain - Extensible
+- **File:** `src/tunacode/tools/authorization/rules.py`
+- **Pattern:** Chain of responsibility - rules evaluated by priority (lower = first)
+- **Existing rules:**
+  | Rule | Priority | Logic |
```

markdownlint-cli2 (0.18.1)

Tables should be surrounded by blank lines

(MD058, blanks-around-tables)